_posts/2017-03-28-comparison-of-regex-engines.md

```diff
@@ -0,0 +1,226 @@
+---
+layout:     post
+title:      A comparison of regex engines
+date:       2017-03-28
+summary:    A performance comparison of regular expression engines including the Rust regex crate.
+categories: regex
+---
+
+{% include chart.js.html %}
+
+# Introduction
+Regular expressions (or just regex) are commonly used in pattern search algorithms. There are many different
+regex engines available with different support of expressions, performance constraints and language bindings.
+Based on the previous work of John Maddock (See his own [regex comparison](http://www.boost.org/doc/libs/1_41_0/libs/regex/doc/gcc-performance.html))
+and the sljit project (See their [regex comparison](http://sljit.sourceforge.net/regex_perf.html))
+I want to give an overview of actively developed engines regarding their performance.
+
+# Test setup
+## Hardware
+The performance is measured on my Dell notebook only. It's not the newest one, but it doesn't matter because
+I used the same hardware for all engines and I'm interested in the performance results compared to each other.
+But here are some hardware information:
+- Chassis: Dell Latitude E7450
+- CPU: Intel® Core™ i5-5300U
+- RAM: 16GB
+- SSD: Samsung PM85 256GB
+
+## Software
+The used software is also not the newest, but newer than the default packages of my installed Ubuntu 16.04:
+- GCC 6.2.0
+- Rustc 1.16.0 and 1.17.0-nightly
+
+I want to get the execution time of each engine for the following regular expression:
```

Comment:
I would like to see here that you not measured the compilation time but the search/match/find processing time.